pkgs/applications/misc/openlp/default.nix

```diff
@@ -0,0 +1,76 @@
+# This file contains all runtime glue: Bindings to optional runtime dependencies
+# for pdfSupport, presentationSupport, and media playback.
+{ stdenv, mkDerivation, wrapGAppsHook, python3Packages
+
+# qt deps
+, qtbase, qttools, qtmultimedia
+
+# optional deps
+, pdfSupport ? false, mupdf ? null, ghostscript ? null
+, presentationSupport ? false, libreoffice-unwrapped ? null, enableVlc ? false
+, gstreamerSupport ? false, gst_all_1 ? null, gstPlugins ? (gst: [
+    gst.gst-plugins-base
+    gst.gst-plugins-good
+    gst.gst-plugins-bad
+    gst.gst-plugins-ugly
+  ])
+
+#, enableMySql ? false      # Untested. If interested, contact maintainer.
+#, enablePostgreSql ? false # Untested. If interested, contact maintainer.
+#, enableJenkinsApi ? false # Untested. If interested, contact maintainer.
+}:
+
+assert pdfSupport -> (mupdf != null || ghostscript != null);
+assert presentationSupport -> libreoffice-unwrapped != null;
+assert enableVlc -> python3Packages.python-vlc or null != null;
+assert gstreamerSupport -> (gst_all_1 != null && gstPlugins gst_all_1 != null);
+
+let
+  # optional packages
+  pdfViewer = if isNull null then mupdf else ghostscript;
```

Comment:
I think you mean
```suggestion
  pdfViewer = if isNull mupdf then mupdf else ghostscript;
```

Reply:
Thanks! Almost: if isNull mpdf then ghostscript… It's fixed and pushed.

Reply:
Oh, yeah, right.